code/modules/projectiles/projectile.dm

```diff
@@ -205,9 +212,101 @@
 	return 1
 
 /obj/item/projectile/proc/OnFired()	//if assigned, allows for code when the projectile gets fired
+	target = get_turf(original)
+	dist_x = abs(target.x - starting.x)
+	dist_y = abs(target.y - starting.y)
+
+	if (target.x > starting.x)
+		dx = EAST
+	else
+		dx = WEST
+
+	if (target.y > starting.y)
+		dy = NORTH
+	else
+		dy = SOUTH
+
+	if(dist_x > dist_y)
+		error = dist_x/2 - dist_y
+	else
+		error = dist_y/2 - dist_x
 	return 1
 
 /obj/item/projectile/proc/process_step()
+	var/sleeptime = 1
+	if(src.loc)
+		if(dist_x > dist_y)
+			if(step_delay)
+				sleep(step_delay)
+			if(kill_count < 1)
+				//del(src)
+				OnDeath()
+				loc = null
+				returnToPool(src)
+				return
+			kill_count--
+			if(error < 0)
+				var/atom/step = get_step(src, dy)
+				if(!step)
+					bullet_die()
+				src.Move(step)
+				error += dist_x
+				bump_original_check()
+				sleeptime = 0//so that bullets going in diagonals don't move twice slower
+			else
+				var/atom/step = get_step(src, dx)
+				if(!step)
+					bullet_die()
+				src.Move(step)
+				error -= dist_y
+				dir = dx
+				if(error < 0)
+					dir = dx + dy
+				bump_original_check()
+		else
+			if(step_delay)

```

Comment:
Stop copypasting 100 lines of code in an if-else statement instead of simply using a proc